packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/ModelTypeProvider.cs

```diff
@@ -15,6 +15,8 @@ public sealed class ModelTypeProvider : TypeProvider
         private readonly InputModelType _inputModel;
         public override string Name { get; }
 
+        public IReadOnlyList<InputModelType> DerivedModels { get; }
```

Comment:
output types should not expose input types on their public surface area.  If we need this property it should be IReadOnlyList<ModelTypeProvider>

I don't see this used anywhere can we remove it?

Reply:
I think when we start to implement discriminator, this might be useful.
And this should be a private readonly field if we keep it now.